core/controllers/question_editor.py

```diff
@@ -0,0 +1,229 @@
+# limitations under the License.
+
+"""Controllers for the questions editor, from where questions are edited and stories
+are created.
+"""
+
+from core.controllers import base
+from core.domain import acl_decorators
+from core.domain import role_services
+from core.domain import question_domain
+from core.domain import question_services
+from core.domain import user_services
+import feconf
+import utils
+
+
+class QuestionEditorPage(base.BaseHandler):
+    """The editor page for a single question."""
+
+    @acl_decorators.can_view_any_question_editor
+    def get(self, question_id):
+        """Handles GET requests."""
+
+        if feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        question = question_services.get_question_by_id(question_id)
+
+        if question is None:
+            raise self.PageNotFoundException(
+                Exception('The question with the given id doesn\'t exist.'))
+
+        self.values.update({
+            'question_id': question['question_id'],
+            'nav_mode': feconf.NAV_MODE_QUESTION_EDITOR
+        })
+
+        self.render_template(
+            'pages/question_editor/question_editor.html',
+            redirect_url_on_logout='/')
+
+
+class EditableQuestionDataHandler(base.BaseHandler):
+    """A data handler for questions which supports writing."""
+
+    def _require_valid_version(self, version_from_payload, question_version):
+        """Check that the payload version matches the given question
+        version.
+        """
+        if version_from_payload is None:
+            raise base.BaseHandler.InvalidInputException(
+                'Invalid POST request: a version must be specified.')
+
+        if version_from_payload != question_version:
+            raise base.BaseHandler.InvalidInputException(
+                'Trying to update version %s of question from version %s, '
+                'which is too old. Please reload the page and try again.'
+                % (question_version, version_from_payload))
+
+    @acl_decorators.can_view_any_question_editor
+    def get(self, question_id):
+        """Populates the data on the individual question page."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        question_domain.Question.require_valid_question_id(question_id)
+
+        question = question_services.get_question_by_id(question_id, strict=False)
+
+        if question is None:
+            raise self.PageNotFoundException(
+                Exception('The question with the given id doesn\'t exist.'))
+
+        self.values.update({
+            'question': question.to_dict()
+        })
+
+        self.render_json(self.values)
+
+    @acl_decorators.can_edit_question
+    def put(self, question_id):
+        """Updates properties of the given question.
+        Also, each change_dict given for editing should have an additional
```

Comment:
This docstring is not required, there is only one type of change object for questions.